photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/NameAndTermFeatureSetContainer.scala

```diff
@@ -14,35 +14,37 @@
  */
 package com.linkedin.photon.ml.data.avro
 
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.{FileSystem, FileUtil, Path}
+import org.apache.spark.rdd.RDD
 import org.apache.spark.SparkContext
 
-import com.linkedin.photon.ml.util.IOUtils
-
 /**
- * A wrapper class for a map of feature section key to a set of [[NameAndTerm]] features
+ * A wrapper class for a map of feature section key to a RDD of [[NameAndTerm]] features
  *
- * @param nameAndTermFeatureSets A [[Map]] of feature section key to [[NameAndTerm]] feature sets
+ * @param nameAndTermFeatureSets A [[Map]] of feature section key to [[NameAndTerm]] feature RDD
  */
-protected[ml] class NameAndTermFeatureSetContainer(nameAndTermFeatureSets: Map[String, Set[NameAndTerm]]) {
+protected[ml] class NameAndTermFeatureSetContainer(nameAndTermFeatureSets: Map[String, RDD[NameAndTerm]]) {
```

Comment:
Owning a `RDD` (or several) should be avoided if possible. Looking at how this class is being used now, I would argue that it should be removed and the `getFeatureNameAndTermToIndexMap`, `saveAsTextFiles`, and `readNameAndTermSetFromTextFiles` functions moved to some utility object (e.g. `FeatureUtils` or an existing alternative).

Reply:
Done